spec/lib/rufo/formatter_source_specs/string_literals.rb.spec

```diff
@@ -38,6 +38,72 @@
 
 "hello #{1} foo"
 
+#~# ORIGINAL single_quote_percent_string_literal_1
+
+'hello #{1} foo'
+
+#~# EXPECTED
+
+"hello \#{1} foo"
```

Comment:
I think it would be better if we were to leave these string untouched. Feels like it would be surprising for users that an escape is added to the string. What do you think?

Reply:
Leaving these strings untouched would also make this consistent with how we treat single quoted strings when it contains a double quote.

Reply:
@gingermusketeer No, this is actually the bug that I'm fixing. `"hello \#{1} foo"` and `'hello #{1} foo'` are the same string, but represented differently. You must escape the `\#{1}`, otherwise Ruby will try to evaluate the interpolation

Reply:
Another way to show this is by running `'hello #{1} foo'` in irb:

```irb
2.5.1 :002 > 'hello #{1} foo'
 => "hello \#{1} foo"
```

Reply:
By untouched I meant that we don't change the string delimiter. 

I think the behaviour we should be trying to achieve is something like (with default config):
 - `"a" => "a"` __No changes__ as it is already correctly formatted.
 - `'b' => "b"` __Replace single with double__
 - `'c\n' => 'c\n'` __No changes__ as this would alter the meaning.
- `'d#{1}' => 'd#{1}'` __No changes__. Could change to `"d\#{1}"`.
- `'"e' => '"e'` __No changes__. Could change to `"\"e"`.

For `d` and `e` I don't think we should make changes as this means that the user will now need to enter escape characters where before this was no necessary. I would guess that one occurrence of a character that requires an escape makes it more likely that there are others, either now or in the future.

The bug in my mind is not that we don't escape the necessary characters when changing the quote style but rather that we change the quote style at all.